File: programasjava/02-EstructurasDeControl/ciclos/Menu.java
import javax.swing.JOptionPane;

public class Menu {
    public static void main(String[] args) {
        String menu="", opcion = "";
        boolean sentinel = true;

        menu = "===== MENU GENERAL =====\n" +
               "1/A)Opción 1\n" +
               "2/B)Opción 2\n" +
               "3/C)Opción 3\n" +
               "4/S)Salir\n" +
               "Elegir Opción: ";
do{
        opcion = JOptionPane.showInputDialog(null, menu);

        switch(opcion.toUpperCase()){
            case "1":
            case "A":
                JOptionPane.showMessageDialog(null,
                        "Opción 1");
            break;

            case "2":
            case "B":
                JOptionPane.showMessageDialog(null,
                        "Opción 2");
                break;

            case "3":
            case "C":
                JOptionPane.showMessageDialog(null,
                        "Opción 3");
                break;

            case "4":
            case "S":
                JOptionPane.showMessageDialog(null,
                        "Salir");
                sentinel = false;
                break;
            default:
                JOptionPane.showMessageDialog(null,
                        "Opción no valida");

        }
    }while(sentinel);
    }
}

File: programaspython/02-estructurasdecontrol/ciclos/menu.py
menu = '''
       ===== MENU GENERAL ===== +
        1/A)Opción 1 +
        2/B)Opción 2 +
        3/C)Opción 3 +
        4/S)Salir +
        Elegir Opción:
        '''

sentinela = True

while(sentinela):
    
    opcion = input(menu).upper()

    if opcion == "1" or opcion == 'A':
        print('Opción 1')
    elif opcion == "2" or opcion == 'B':
        print('Opción 2')
    elif opcion == "3" or opcion == 'C':
        print('Opción 3')
    elif opcion == "4" or opcion == 'S':
        print('Salir')
        sentinela = False
    else:
        print('Opción no valida')

print("El programa ha terminado")


